service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/DiagnosticReportMapper.java

```diff
@@ -73,13 +75,18 @@ public class DiagnosticReportMapper {
     private final ConfidentialityService confidentialityService;
 
     public String mapDiagnosticReportToCompoundStatement(DiagnosticReport diagnosticReport) {
-        List<Specimen> specimens = fetchSpecimens(diagnosticReport);
         List<Observation> observations = fetchObservations(diagnosticReport);
+        List<Specimen> specimens = fetchSpecimens(diagnosticReport, observations);
         final IdMapper idMapper = messageContext.getIdMapper();
         markObservationsAsProcessed(idMapper, observations);
+        observations = assignDummySpecimensToObservationsWithNoSpecimen(observations, specimens);
```

Comment:
Given that this method does a bunch of gymnastics to filter out Filing Comment Observations, why not move it down a couple of lines and call it on `observationsExcludingFilingComments` ?

Reply:
Good suggestion, made the change.